### PR TITLE
fix: typography component updates

### PR DESCRIPTION
**Pull request by tibuurcio**
## Summary

This PR updates the Typography components to better reflect our current customization needs coming from the platforms. It does the following:
- Changes the `size` prop to generically accept numbers rather then only `'base' | 'sm' | 'lg' | 'xl'`
- Makes the `color` prop work (it wasn't) and provide some auto-complete suggestions for it based on `ColorText` design tokens.
- Adds a new `tooltip` prop to both `Typography.Text` and `Typography.Link` to be used when those components are used inside a tooltip. Also adds a story for it.
- Re-organize all Typography component related stories, essentially moving individual prop stories into the main one and making them toggleable through controls.

## Testing Plan

- [X] Was this tested locally?
- [] Test in the platforms

## Reference Issue (For mParticle employees only. Ignore if you are an outside contributor)

- Closes https://go.mparticle.com/work/UNI-1151

### CODE REVIEW

File: src/components/general/Typography/Typography.tsx [hunk 1]
```diff
@@ -1,37 +1,23 @@
...
 
 // TODO: Replace hardcoded values in getFontSize and getLineHeight with tokens when design is ready
 // These values are currently coming from https://www.figma.com/design/LffDbOUjeYqDMZ3djs9Cga/mParticle-Foundation-v1.0.1?node-id=3745-8164&m=dev
 const getFontSize = (size: TypographySize): number => {
   if (size === 'base') return 14
   if (size === 'sm') return 12
   if (size === 'lg') return 16
-  return 20
+  if (size === 'xl') return 20
+  return size
```
Comment (danielsiemens) [inline src/components/general/Typography/Typography.tsx:20]: NIce!

**Review by danielsiemens — APPROVED** [review]

File: src/components/general/Typography/Link.stories.tsx [hunk 2]
```diff
@@ -1,35 +1,74 @@
...
 const meta: Meta<typeof Typography.Link> = {
-  title: 'Components/General/Link',
+  title: 'Components/General/Typography/Link',
   component: props => <Typography.Link {...props}>Example Link</Typography.Link>,
 
   args: {
+    children: 'Example Text',
+    type: undefined,
+    color: undefined,
+    size: 'base',
```
Comment (nastyastavitskaya) [inline src/components/general/Typography/Link.stories.tsx:16]: hooray! 🎉

File: src/components/general/Typography/Typography.tsx (hunk 1 above)
Comment (nastyastavitskaya) [reply src/components/general/Typography/Typography.tsx:20]: I am wondering if we need `xs` = 10px 🤔

File: src/components/general/Typography/Typography.tsx [hunk 3]
```diff
@@ -41,50 +27,106 @@ const getLineHeight = (size: TypographySize): number => {
...
-  const fontSize = getFontSize(size)
-  const lineHeight = getLineHeight(size)
+interface InternalTypographyProps {
+  size?: TypographySize
+  color?: TypographyColor
+  children: ReactNode
+}
+
+export const Typography = () => {
+  return <>DO NOT USE</>
```
Comment (nastyastavitskaya) [inline src/components/general/Typography/Typography.tsx:37]: ...or you'll be fired 😝

**Review by nastyastavitskaya — APPROVED** [review]
this is an incredible improvement! 👏

**Pull request merged**